internal/bufferedwrites/upload_handler.go

```diff
@@ -171,6 +171,29 @@ func (uh *UploadHandler) Finalize() (*gcs.MinObject, error) {
 	return obj, nil
 }
 
+// Flush uploads any data in the write buffer.
+func (uh *UploadHandler) Flush() (int64, error) {
+	uh.wg.Wait()
+
+	if uh.writer == nil {
```

Comment:
can you extract this into common method like ensureWriter and reuse it in finalize() & upload methods.